queries.sql

```diff
@@ -68,7 +71,99 @@ SELECT AVG(weight_kg) FROM animals;
 
 /* What is the average number of escape attempts per animal type of those born between 1990 and 2000? */
-SELECT AVG(escape_attempts), species_id  FROM animals WHERE date_of_birth > '1990-01-01' AND date_of_birth < '2000-01-01' GROUP BY species_id ;
\ No newline at end of file
+SELECT AVG(escape_attempts), species_id  FROM animals WHERE date_of_birth > '1990-01-01' AND date_of_birth < '2000-01-01' GROUP BY species_id ;
+
+
+
+
+
+
+/* Tasks-3*/ 
+
+/* What animals belong to Melody Pond? */
+SELECT * FROM animals INNER JOIN owners ON owner_id = owners.id WHERE full_name = 'Melody Pond';
+
+/* List of all animals that are pokemon (their type is Pokemon). */
+SELECT * FROM animals INNER JOIN species ON species.id = animals.species_id WHERE species.name = 'Pokemon';
+
+/* List all owners and their animals, remember to include those that don't own any animal. */
+SELECT full_name, name  FROM owners LEFT JOIN animals ON animals.owner_id = owners.id ;
+
+/*  How many animals are there per species?  */ 
+SELECT species.name, COUNT(*) FROM animals FULL OUTER JOIN species ON animals.species_id = species.id GROUP BY species.id;
+
+/* List all Digimon owned by Jennifer Orwell. */
+SELECT animals.name, species.name FROM animals 
+INNER JOIN owners 
+ON  owners.id = animals.owner_id
+INNER JOIN species 
+ON  species.id = animals.species_id
+WHERE owners.full_name = 'Jennifer Orwell'
+AND species.name = 'Digimon';
+
+/* List all animals owned by Dean Winchester that haven't tried to escape.  */
+SELECT animals.name FROM animals
+LEFT JOIN owners 
+ON owners.id = animals.owner_id
+WHERE animals.escape_attempts = 0
+AND owners.full_name = 'Dean Winchester';
+
+/* Who owns the most animals? */
+SELECT full_name, COUNT(*) as total FROM owners
+LEFT JOIN animals 
+ON owners.id = animals.owner_id GROUP BY full_name ORDER BY total desc;
+
+
+/* Tasks-4*/ 
+/*  Who was the last animal seen by William Tatcher? */
+SELECT animal_name FROM visits WHERE vet_name = 'William Tatcher' ORDER BY date_of_visit DESC LIMIT 1;
+
+/*  How many different animals did Stephanie Mendez see?  */
+SELECT DISTINCT animal_name FROM visits WHERE vet_name = 'Stephanie Mendez';
+
+/*  List all vets and their specialties, including vets with no specialties.  */
+SELECT name, species_name FROM specializations LEFT JOIN vets ON specializations.vet_name = name ;
+
+/*  List all animals that visited Stephanie Mendez between April 1st and August 30th, 2020.  */
+SELECT DISTINCT name  FROM visits JOIN animals ON 
+visits.vet_name = 'Stephanie Mendez'
+AND visits.date_of_visit > '2020-04-01' AND visits.date_of_visit < '2020-08-30' ;
+
+/*  What animal has the most visits to vets?  */
+SELECT DISTINCT animals.name, COUNT(animal_name)  FROM visits JOIN animals ON 
+visits.animal_name = animals.name
+GROUP BY animals.name ORDER BY COUNT(animal_name) DESC LIMIT 1;
+
+/*  Who was Maisy Smith's first visit?  */
+SELECT animal_name, date_of_visit FROM visits ORDER BY date_of_visit ASC LIMIT 1;
+
```

Comment:
- [ ] This query is not displaying, who was Maisy Smith's first visit. You should have written in your solution `WHERE vets.name = 'Maisy Smith' `

Reply:
Thanks a lot for your time, I really appreciate it. 
I have already solved this. :heavy_check_mark:

Reply:
Now It looks like this :
```cmd 
SELECT animal_name, date_of_visit FROM visits WHERE vet_name = 'Maisy Smith'  ORDER BY date_of_visit   ASC
 LIMIT 1;